dali/kernels/reduce/mean_stddev_gpu_impl.cuh

```diff
@@ -57,21 +57,11 @@ class MeanImplBase {
 
   using scale_t = typename Postprocessor::scale_t;
 
-  Postprocessor *GetPostprocessorsImpl(WorkArea &wa) const {
-    assert(!This().ReduceBatch());
-    int n = This().SimplifiedOutputShape().num_samples();
-    Postprocessor *pp = wa.ParamBuffer<Postprocessor>(n);
-    for (int i = 0; i < n; i++) {
-      DALI_ENFORCE(This().ReducedElements(i) > 0, "Cannot calculate a mean from 0 elements");
-      pp[i].scale = 1.0 / This().ReducedElements(i);
-    }
-    return pp;
-  }
-
-  Postprocessor GetPostprocessorImpl() const {
-    assert(This().ReduceBatch());
-    DALI_ENFORCE(This().TotalReducedElements() > 0, "Cannot calculate a mean from 0 elements");
-    return { static_cast<scale_t>(1.0 / This().TotalReducedElements()) };
+  Postprocessor GetPostprocessorImpl(int sample_index, bool reduce_batch) const {
+    int64_t reduced_elems = reduce_batch ? This().TotalReducedElements()
+                                         : This().ReducedElements(sample_index);
+    DALI_ENFORCE(reduced_elems > 0, "Cannot calculate a mean from 0 elements");
```

Comment:
I still have a feeling, that those Postprocessors should just get the number of reduced elements, but maybe there will be some sample-specific behaviour in the future.